environment.m

```diff
@@ -0,0 +1,44 @@
+function reward = environment(action,connectivity,avg_msgs,sigma2)
+
+    num_UE = length(connectivity) - 1;
+    
+    % make sure inputs are row-vectors
+    action = reshape(action,[1,length(action)]);
+    avg_msgs = reshape(avg_msgs,[1,length(avg_msgs)]);
+
+    % find number of messages sent at this time-step
+    messages = generate_messages(avg_msgs,sigma2);
+    
+    % select heads, including base-station
+    heads = [1, find(action) + 1];          % offset for base-station

```

Comment:
- `find(action)` will output indices of 0 elements.
- should probably be `find(action==1)`